### PR TITLE
Feat/text search

### LINKED ISSUE #3050: text search missing
Hyper doesn't have a default "search" function built-in and the `hyper-search` plugin has issues and doesn't work with the latest hyper 2.0.0.

### PR DESCRIPTION
**Pull request by El-Dringo-Brannde**
Closes #3050 

Adds text search natively within the Hyper Terminal, similar to [hyper-search](https://github.com/jaanauati/hyper-search). Except this uses the [xterm search addon](https://github.com/xtermjs/xterm.js/blob/master/src/addons/search/search.ts), and not a home rolled solution. 

Tested on Mac only, hotkey mapping should support Linux and Windows as well. 

_To Test_ 
Start up locally, and open up the `Edit` menu, and select `Search`, or press `Command + F` to toggle the search box pop up. From there it is just searching for the word you want to use, and move up and down the search results.

### CODE REVIEW

**Comment by El-Dringo-Brannde** [comment]
@nrempel You could use my fork for the time being? I have no other changes apart from the little search box. You can do a git diff between the two repos and see for yourself

**Comment by heubergen** [comment]
Anything new about this? @chabou

**Comment by kevinschaich** [comment]
In addition to #3050 as mentioned above, this would also close #140 and #541, which have both been open since 2016. 

@chabou @Tyriar @jcrben @matheuss @albinekb tagging you guys since you've contributed to those three issues addressing a search feature, can we get a +1 on this please? Author seems to have addressed the comments mentioned in the original review by @ppot.

Thanks for this @El-Dringo-Brannde! Search has been the only thing stopping me from switching to Hyper.

File: app/keymaps/darwin.json [hunk 1]
```diff
@@ -36,6 +36,7 @@
   "editor:copy": "command+c",
   "editor:paste": "command+v",
   "editor:selectAll": "command+a",
+  "editor:search": "command+f",
```
Comment (kevinschaich) [inline app/keymaps/darwin.json]: Might also be nice to bind ` Escape ` to close the search box.

Comment (El-Dringo-Brannde) [reply app/keymaps/darwin.json]: Whoops, I've been away for quite some time this took off <3. I'll try and make the changes later tonight.

Comment (El-Dringo-Brannde) [reply app/keymaps/darwin.json]: Well.. Apparently I 'later tonight' to me is almost a month later, but have no fear the changes have been implemented in b978a3. 

@kevinschaich @chabou @ppot rereview?

**Comment by neil-orans** [comment]
Status on this review?

**Comment by Sufiane** [comment]
Hi all, Came for the status of this too, what's up? :)

**Comment by El-Dringo-Brannde** [comment]
@chabou Any status update for us?

**Pull request merged**